lua/roslyn/commands.lua

```diff
@@ -5,109 +5,125 @@ local M = {}
-                for _, buffer in ipairs(attached_buffers) do
-                    if vim.api.nvim_buf_is_valid(buffer) then
-                        vim.api.nvim_exec_autocmds("FileType", { group = "Roslyn", buffer = buffer })
-                    end
-                end
-                if remove_listener then
-                    remove_listener()
-                end
-            end
-
-            remove_listener = roslyn_emitter:on("stopped", restart_lsp)
-
-            local force_stop = vim.loop.os_uname().sysname == "Windows_NT"
-            client.stop(force_stop)
-        end,
-    },
-    stop = {
-        impl = function()
-            local client = vim.lsp.get_clients({ name = "roslyn" })[1]
-            if not client then
-                return
-            end
-
-            -- TODO: Change this to `client:request` when minimal version is `0.11`
-            ---@diagnostic disable-next-line: missing-parameter, param-type-mismatch
-            client.stop(true)
-        end,
-    },
-    target = {
-        impl = function()
-            local bufnr = vim.api.nvim_get_current_buf()
-            local root = vim.b.roslyn_root or require("roslyn.sln.utils").root(bufnr)
-
-            local roslyn_lsp = require("roslyn.lsp")
-
-            local targets = vim.iter({ root.solutions, root.solution_filters }):flatten():totable()
-            vim.ui.select(targets or {}, { prompt = "Select target solution: " }, function(file)
-                if not file then
-                    return
-                end
-
-                vim.lsp.stop_client(vim.lsp.get_clients({ name = "roslyn" }), true)
-                local sln_dir = vim.fs.dirname(file)
-                roslyn_lsp.start(bufnr, assert(sln_dir), roslyn_lsp.on_init_sln(file))
-            end)
-        end,
-    },
+	restart = {
+		impl = function()
+			local client = vim.lsp.get_clients({ name = "roslyn" })[1]
+			if not client then
+				return
+			end
+
+			local attached_buffers = vim.tbl_keys(client.attached_buffers)
+
+			---@type function | nil
+			local remove_listener = nil
+
+			local function restart_lsp()
+				for _, buffer in ipairs(attached_buffers) do
+					if vim.api.nvim_buf_is_valid(buffer) then
+						vim.api.nvim_exec_autocmds("FileType", { group = "Roslyn", buffer = buffer })
+					end
+				end
+				if remove_listener then
+					remove_listener()
+				end
+			end
+
+			remove_listener = roslyn_emitter:on("stopped", restart_lsp)
+
+			local force_stop = vim.loop.os_uname().sysname == "Windows_NT"
+			client.stop(force_stop)
+		end,
+	},
+	stop = {
+		impl = function()
+			local client = vim.lsp.get_clients({ name = "roslyn" })[1]
+			if not client then
+				return
+			end
+
+			-- TODO: Change this to `client:request` when minimal version is `0.11`
+			---@diagnostic disable-next-line: missing-parameter, param-type-mismatch
+			client.stop(true)
+		end,
+	},
+	target = {
+		impl = function()
+			select_sln_and_start_lsp()
```

Comment:
I want `target` to stay a little bit different from start, because otherwise it doesn't seem necessary to create a `Roslyn start` command.

I think `target` should still need interaction even if there is only a single target, if we are going to add `start`